swift/proxy/controllers/base.py

```diff
@@ -839,20 +839,22 @@ def get_working_response(self, req):
         source, node = self._get_source_and_node()
         res = None
         if source:
-            res = Response(request=req)
+            res = Response(request=req, status=source.status)
             if req.method == 'GET' and \
                     source.status in (HTTP_OK, HTTP_PARTIAL_CONTENT):
                 res.app_iter = self._make_app_iter(req, node, source)
                 # See NOTE: swift_conn at top of file about this.
                 res.swift_conn = source.swift_conn
-            res.status = source.status
```

Comment:
The reason for this is that swob Response __init__  tries to assign default content-type as 'text/html; charset=UTF-8', with this change, if we pass the 304 to Response __init__ it will remove the content-type & content-length from the headers.